src/ApiClient.ts

```diff
@@ -298,6 +317,81 @@ export default class ApiClient {
     }
   }
 
+  // Get first pageSize envelopes for a number of different topics, intended to efficiently
+  // use the batch-query API in a targeted manner. Returns a map of topic to message envelopes
+  async batchQuery(
+    { contentTopics, startTime, endTime }: QueryParams,
```

Comment:
It feels a little funny to me that this method expects all queries in the batch to have the same `startTime`, `endTime`, `pageSize`, and `sortDirection`. That works fine for the current use-case, and likely for the next one, but give that this is the only exposed method to access the `batchQuery` endpoint I wonder if we want to be so prescriptive.

Reply:
I can see that, especially with the defaults the common case of identical params should still be pretty simple. I'll make the change to take a list of QueryParams

Reply:
I think I'll have to change the return type from `Promise<Map<string, envelopes[]>>` to just `Promise<envelopes[][]>`. The idea is that `batchQuery` just takes a list of normal queries and acts as `query` would.

Reply:
Ya. I think we can take this higher level abstraction of assuming the queries have the same parameters and just put it elsewhere.